### PR TITLE
pumpkin fix and sprite improvement

### PR DESCRIPTION
**Pull request by alsoandanswer**
KObrAp70Nd

and improves sprites for pumpkin and candles.

fixes drop sounds for some boxes too.

### CODE REVIEW

File: code/game/objects/items/weapons/candle.dm [hunk 1]
```diff
@@ -38,22 +39,26 @@
...
+		playsound(src.loc, 'sound/items/cigs_lighters/cig_light.ogg', 50, 1)
 		//src.damtype = "fire"
 		for(var/mob/O in viewers(usr, null))
 			O.show_message(flavor_text, 1)
 		set_light(CANDLE_LUM)
+		update_icon()
 		START_PROCESSING(SSprocessing, src)
 
-/obj/item/weapon/flame/candle/process()
+/obj/item/weapon/flame/candle/process(mob/user as mob)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/candle.dm]:
```suggestion
/obj/item/weapon/flame/candle/process(mob/user)
```

File: code/game/objects/items/weapons/candle.dm [hunk 2]
```diff
@@ -38,22 +39,26 @@
...
 		return
+	update_icon()
 	wax--
 	if(!wax)
-		new/obj/item/trash/candle(src.loc)
+		new /obj/item/trash/candle(src.loc)
 		if(istype(src.loc, /mob))
 			src.dropped()
-
+		to_chat(user, span("notice", "The candle burns out."))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/candle.dm]: Who is user here exactly?

File: code/modules/clothing/head/misc_special.dm [hunk 3]
```diff
@@ -139,16 +139,96 @@
...
+		M.update_inv_head(0)
+		M.update_inv_l_hand(0)
+		M.update_inv_r_hand(1)
+
+/obj/item/clothing/head/pumpkin/lantern/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
+	if(W.iswelder())
+		var/obj/item/weapon/weldingtool/WT = W
+		if(WT.isOn()) //Badasses dont get blinded by lighting their candle with a welding tool
+			light("<span class='notice'>\The [user] casually lights the [name] with [W].</span>")
```
Comment (NonQueueingMatt) [inline code/modules/clothing/head/misc_special.dm]:
```suggestion
			light("<span class='notice'>\The [user] casually lights [name] with [W].</span>")
```

File: code/modules/clothing/head/misc_special.dm [hunk 4]
```diff
@@ -139,16 +139,96 @@
...
+		var/obj/item/weapon/flame/candle/C = W
+		if(C.lit)
+			light()
+
+/obj/item/clothing/head/pumpkin/lantern/proc/light(var/flavor_text = "<span class='notice'>\The [usr] lights the [name].</span>")
+	if(!src.lit)
+		src.lit = TRUE
+		playsound(src.loc, 'sound/items/cigs_lighters/cig_light.ogg', 50, 1)
+		//src.damtype = "fire"
+		for(var/mob/O in viewers(usr, null))
```
Comment (NonQueueingMatt) [inline code/modules/clothing/head/misc_special.dm]: Why is this not a visible message though

File: code/modules/clothing/head/misc_special.dm [hunk 5]
```diff
@@ -139,16 +139,96 @@
...
+			O.show_message(flavor_text, 1)
+		set_light(CANDLE_LUM)
+		update_icon()
+		update_clothing_icon()
+		START_PROCESSING(SSprocessing, src)
+
+/obj/item/clothing/head/pumpkin/lantern/process(mob/user as mob)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/head/misc_special.dm]:
```suggestion
/obj/item/clothing/head/pumpkin/lantern/process(mob/user)
```

Same shit, who is supposed to be user here

File: code/modules/clothing/head/misc_special.dm (hunk 4 above)
Comment (alsoandanswer) [reply code/modules/clothing/head/misc_special.dm]: it's in the light() proc as var/flavor_text =

File: code/modules/clothing/head/misc_special.dm (hunk 5 above)
Comment (alsoandanswer) [reply code/modules/clothing/head/misc_special.dm]: so people can see in chat if a candle snuffed out on their screen

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/items/weapons/candle.dm [hunk 6]
```diff
@@ -27,7 +28,7 @@
 	if(W.iswelder())
 		var/obj/item/weapon/weldingtool/WT = W
 		if(WT.isOn()) //Badasses dont get blinded by lighting their candle with a welding tool
-			light("<span class='notice'>\The [user] casually lights the [name] with [W].</span>")
+			light("<span class='notice'>\The [user] casually lights \the [name] with [W].</span>")
```
Comment (Arrow768) [inline code/game/objects/items/weapons/candle.dm]: `span()`

File: code/game/objects/items/weapons/candle.dm [hunk 7]
```diff
@@ -36,24 +37,28 @@
 			light()
 
 
-/obj/item/weapon/flame/candle/proc/light(var/flavor_text = "<span class='notice'>\The [usr] lights the [name].</span>")
+/obj/item/weapon/flame/candle/proc/light(var/flavor_text = "<span class='notice'>\The [usr] lights \the [name].</span>")
```
Comment (Arrow768) [inline code/game/objects/items/weapons/candle.dm]: `span()`

File: code/modules/clothing/head/misc_special.dm [hunk 8]
```diff
@@ -139,16 +139,98 @@
...
+		M.update_inv_head(0)
+		M.update_inv_l_hand(0)
+		M.update_inv_r_hand(1)
+
+/obj/item/clothing/head/pumpkin/lantern/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
+	if(W.iswelder())
+		var/obj/item/weapon/weldingtool/WT = W
+		if(WT.isOn()) //Badasses dont get blinded by lighting their candle with a welding tool
+			light("<span class='notice'>\The [user] casually lights \the [name] with [W].</span>")
```
Comment (Arrow768) [inline code/modules/clothing/head/misc_special.dm]: `span()`

**Pull request merged**